tests/SbrpTests/ValidationTests.cs

```diff
@@ -0,0 +1,166 @@
+
+        foreach (var package in packages)
+        {
+            bool isSigned = await IsPackageSignedAsync(package, verifier, settings);
+            Assert.False(isSigned, $"{package} is signed. Signed packages are not allowed in {SbrpRepoIdentifier}.");
+        }
+    }
+
+    private string[] GetPackages()
+    {
+        string buildPackagesDirectory = PathUtilities.GetSourceBuildPackagesShippingDir();
+
+        string[] packages = Directory.GetFiles(buildPackagesDirectory, "*.nupkg", SearchOption.AllDirectories);
+
+        if (packages.Length == 0)
+        {
+            throw new FileNotFoundException($"No packages found in {buildPackagesDirectory}");
+        }
+        return packages;
+    }
+
+    private bool HasSbrpAttribute(MetadataReader reader) =>
+        reader.CustomAttributes
+            .Select(attrHandle => reader.GetCustomAttribute(attrHandle))
+            .Any(attr => IsAttributeSbrp(reader, attr));
+
+    private bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
+    {
+        string attributeType = string.Empty;
+
+        if (attr.Constructor.Kind == HandleKind.MemberReference)
+        {
+            var mref = reader.GetMemberReference((MemberReferenceHandle)attr.Constructor);
+            if (mref.Parent.Kind == HandleKind.TypeReference)
+            {
+                var tref = reader.GetTypeReference((TypeReferenceHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tref.Namespace)}.{reader.GetString(tref.Name)}";
+            }
+            else if (mref.Parent.Kind == HandleKind.TypeDefinition)
+            {
+                var tdef = reader.GetTypeDefinition((TypeDefinitionHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+            }
+        }
+        else if (attr.Constructor.Kind == HandleKind.MethodDefinition)
+        {
+            var mdef = reader.GetMethodDefinition((MethodDefinitionHandle)attr.Constructor);
+            var tdef = reader.GetTypeDefinition(mdef.GetDeclaringType());
+            attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+        }
+
+        if (attributeType == SbrpAttributeType)
+        {
+            var decodedValue = attr.DecodeValue(DummyAttributeTypeProvider.Instance);
+            try
+            {
+                return decodedValue.FixedArguments[0].Value?.ToString() == "source" && decodedValue.FixedArguments[1].Value?.ToString() == SbrpRepoIdentifier;
+            }
+            catch
+            {
+                throw new InvalidOperationException($"{SbrpAttributeType} is not formatted properly with a key, value pair.");
+            }
+        }
+
+        return false;
+    }
+
+    private async Task<bool> IsPackageSignedAsync(string packagePath, PackageSignatureVerifier verifier, SignedPackageVerifierSettings settings)
```

Comment:
Check for any other methods that can be `static` as well.

```suggestion
    private static async Task<bool> IsPackageSignedAsync(string packagePath, PackageSignatureVerifier verifier, SignedPackageVerifierSettings settings)
```